src/sentry/integrations/vsts/search.py

```diff
@@ -0,0 +1,39 @@
+from __future__ import absolute_import
+
+from sentry.api.bases.organization import OrganizationEndpoint, OrganizationPermission
+from rest_framework.response import Response
+
+from sentry.models import Integration
+
+
+class VstsSearchEndpoint(OrganizationEndpoint):
+    permission_classes = (OrganizationPermission, )
+
+    def get(self, request, organization, integration_id):
+        try:
+            integration = Integration.objects.get(
+                organizations=organization,
+                id=integration_id,
+                provider='vsts',
+            )
+        except Integration.DoesNotExist:
+            return Response(status=404)
+
+        field = request.GET.get('field')
+        query = request.GET.get('query')
+        if field is None:
+            return Response({'detail': 'field is a required parameter'}, status=400)
+        if not query:
+            return Response({'detail': 'query is a required parameter'}, status=400)
+
+        installation = integration.get_installation(organization.id)
+
+        if field == 'externalIssue':
+            if not query:
+                return Response([])
+
+            resp = installation.get_client().search_issues(integration.name, query)
+            return Response([{
+                'label': '(%s) %s' % (i['fields']['system.id'], i['fields']['system.title']),
+                'value': i['fields']['system.id'],
+            } for i in resp.get('results', [])])
```

Comment:
you need to make sure you're returning a response at the end in case your conditionals aren't hit -- just `return Response(status=400)` or something similar is fine. otherwise this will 500